CARDS-2118: Add an Unsubscribe link in the footer + CARDS-2119: The unsubscribe page looks sloppy

marta- opened the pull request:
**Includes:**
* [CARDS-2118](https://phenotips.atlassian.net/browse/CARDS-2118) - Patient portal: add an Unsubscribe link in the footer
* [CARDS-2119](https://phenotips.atlassian.net/browse/CARDS-2119) - Patient portal: the unsubscribe page looks sloppy

**Testing:**
* start `prems`
* go to `/Survey.html` without a token - there should be no Unsubscribe link (it only appears when the patient is authenticated)
* as admin, create a Patient information form (can be left empty) and a Visit information form with any survey
* as patient, go to the link for filling out that survey -> notice the unsubscribe link in the footer
* click on Unsubscribe and follow instructions
* once unsubscribed, open the Patient information form and check that the "unsubscribed" answer is set accordingly
* bonus: start in `proms` and do similar testing

**Reviewing:**
* commit by commit, ignoring whitespace recommended

**Preview:**
![image](https://user-images.githubusercontent.com/651980/221086166-5c9394f9-3a3e-4933-a988-5e3b9e9433c1.png)

![image](https://user-images.githubusercontent.com/651980/221086101-8e5006c2-4cce-469b-bfa4-0f40590c6e11.png)


[CARDS-2118]: https://phenotips.atlassian.net/browse/CARDS-2118?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CARDS-2119]: https://phenotips.atlassian.net/browse/CARDS-2119?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

marta- committed:
CARDS-2118 - Patient portal: add an Unsubscribe link in the footer

marta- committed:
CARDS-2119 - Patient portal: the unsubscribe page looks sloppy

sashaandjic reviewed (approved):
tested
-----

**Unsubscribe at the bottom of the authenticated page:**
![image](https://user-images.githubusercontent.com/23243244/221117749-ce8e46e7-5f73-4ae1-90fa-d167f61a598a.png)

**After clicking on the `Unsubscribe` link:**
![image](https://user-images.githubusercontent.com/23243244/221117940-11ff1377-4baa-465b-b2a6-45ea31b96edf.png)

After clicking on the `UNSUBSCRIBE` button:
![image](https://user-images.githubusercontent.com/23243244/221118061-f6729690-089f-4ff6-9a9d-1170a9ce7153.png)

**The value of the `The patient unsubscribed from all email notifications` is set to `Yes`**
![image](https://user-images.githubusercontent.com/23243244/221119126-c1963976-98da-446e-a621-2c2733fe539d.png)

**You can resubscribe:**
![image](https://user-images.githubusercontent.com/23243244/221118634-6c15947e-6b13-4b8e-bf5a-6e8551b99235.png)

**And the value of the `The patient unsubscribed from all email notifications` is set to `No`:**
![image](https://user-images.githubusercontent.com/23243244/221119083-31f1d44a-dfaf-49e7-9c70-264aa343a340.png)

sashaandjic commented:
Question: I think moving `Unsubscribe` link below TOU would look better. WDYT?
![image](https://user-images.githubusercontent.com/23243244/221119818-f4685090-4237-444c-83e6-4c1e6e7e4c8b.png)

marta- commented:
> Question: I think moving `Unsubscribe` link below TOU would look better. WDYT?

I'm not a fan, especially on Form & Review pages - I think it's a waste of vertical space for two very rarely used links that fit just fine on one line.

![image](https://user-images.githubusercontent.com/651980/221207003-a6b9c9f0-fbf5-438a-8641-5b205e74e0bc.png)
![image](https://user-images.githubusercontent.com/651980/221207171-7af27765-d00b-4321-97bc-efd496b338fc.png)
![image](https://user-images.githubusercontent.com/651980/221207682-7a732d07-2837-4d86-a830-37eafadc1234.png)

marta- committed:
CARDS-2118 - Patient portal: add an Unsubscribe link in the footer
Addressing comments: make sure the footer does not display a separator when the unsubscribe link is absent (prems)

sdumitriu reviewed (commented):
In proms mode there ar a few issues with this:

- one can access the patient portal without a token, authenticate using the personal information, and then in theory they will be authenticated, but the Unsubscribe link doesn't show up
- when accessing without a token, the bottom breadcrumbs still has two items, but one is empty, so there's an extra separator dot at the end
- when accessing with a token, the unsubscribe link can be used before the patient actually validates who they are; that's not really bad, since we already have an unsubscribe link in the email that doesn't require authenticating, but from the user's perspective it may seem wrong that they can do something with their account before they actually "log in" into the account

In prems mode things are much better, the only tiny bug is when trying to access without a token, the extra dot at the end of the breadcrumbs is present, moving the ToU link a bit off center.

marta- committed:
CARDS-2118 - Patient portal: add an Unsubscribe link in the footer
Addressing comments: better alignment of footer links

marta- commented:
> In proms mode there ar a few issues with this

Since `prems` is active and `proms` isn't, I suggest we make a separate task for `proms` specific fixes so we can roll out the prems improvements asap.
 
> In prems mode things are much better, the only tiny bug is when trying to access without a token, the extra dot at the end of the breadcrumbs is present, moving the ToU link a bit off center.

Addressed in 0613876.

marta- commented:
> * one can access the patient portal without a token, authenticate using the personal information, and then in theory they will be authenticated, but the Unsubscribe link doesn't show up
> * when accessing with a token, the unsubscribe link can be used before the patient actually validates who they are; that's not really bad, since we already have an unsubscribe link in the email that doesn't require authenticating, but from the user's perspective it may seem wrong that they can do something with their account before they actually "log in" into the account

Actually the more I think about all this the more it make it makes sense the way it is now even for `proms`. "Unsubscribe" is there because the patient was brought to that screen from their email link. There's no reason to want to unsubscribe if you came here out of your own free will. Anyway, I created https://phenotips.atlassian.net/browse/CARDS-2128 and we can deal with one way or the other it when time comes to wrap up `proms`.

sdumitriu reviewed (commented):
The ToU link is actually a button, while the Unsubscribe is a link. That makes the two not vertically in line (ToU is 2 pixels lower).

sdumitriu commented:
> Actually the more I think about all this the more it make it makes sense the way it is now even for proms. "Unsubscribe" is there because the patient was brought to that screen from their email link. There's no reason to want to unsubscribe if you came here out of your own free will. 

You're right.

marta- commented:
> The ToU link is actually a button, while the Unsubscribe is a link. That makes the two not vertically in line (ToU is 2 pixels lower).

The look pretty well-aligned to me. This is in FF, same in chrome:

![image](https://user-images.githubusercontent.com/651980/221441081-a0703b15-9395-4963-adce-98e9feb98f03.png)

marta- commented:
> The ToU link is actually a button, while the Unsubscribe is a link. That makes the two not vertically in line (ToU is 2 pixels lower)

Replicated and addressed in f33f1ea.
